### ISSUE TITLE
plugins/staticheaders: Add a `Vary: Cookie` header by default

### ISSUE THREAD

**Issue by empijei** [google/go-safeweb#235]
This is important for XSRF double-submit cookies and for other [potential non-security related breakages](https://drupal.stackexchange.com/questions/25271/what-does-the-vary-cookie-header-actually-do-for-serving-pages-from-varnish).

**Comment by kele** [comment]
MDN: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Vary.

I agree that for now we should add the value to the `Vary` header, but we might consider actually having a separate plugin to control `Vary` (or other headers like the ones for Cookies) and expose safe mutators.

**Comment by empijei** [comment]
Vary should probably be treated in a append-only way, and we might want to even consider adding this to the framework rather than try to have a plugin that is shared across multiple plugins.

**Comment by kele** [comment]
The issue with having this in the core of the framework rather than a plugin is harder migration, because we usually don't offer opt-outs from core features. That being said, I'm not sure it's a big concern.

I'd be in favor of having a separate API for dealing with the `Vary` header, similarly to how we have a separate one for cookies.

**Comment by empijei** [comment]
What is the alternative? Add a generic way to support append-only headers?

**Comment by kele** [comment]
No. Have a plugin that claims `Vary` and offers an API to append to it -- this is one way to go about it.
